scaladoc/test-source-links/dotty/tools/scaladoc/source-links/RemoteLinksTest.scala

```diff
@@ -38,16 +38,17 @@ class RemoteLinksTest:
   def runTest =
     assertTrue(mtslAll.nonEmpty)
     val mtsl = randomGenerator.shuffle(mtslAll).take(80) // take 80 random entries
-    val pageToMtsl: Map[String, List[(String, String)]] = mtsl.groupMap(_._2.split("#L").head)(v => (v._1, v._2.split("#L").last))
-    pageToMtsl.foreach { case (link, members) =>
+    val pageToMtsl: Map[String, List[(String, String)]] =
+      mtsl.groupMap(_._2.split("#L").head)(v => (v._1, v._2.split("#L").last.toInt))
+    pageToMtsl.toSeq.sortBy(_._1).foreach { case (link, members) =>
       try
         val doc = getDocumentFromUrl(link)
         println(s"Checking $link")
         members.foreach { case (member, expectedLine) =>
           if !member.startsWith("given_") then // TODO: handle synthetic givens, for now we disable them from testing
             val toLine = expectedLine + 3
             val memberToMatch = member.replace("`", "")
-            val lineCorrectlyDefined = (expectedLine.toInt until toLine.toInt).exists{ line =>
+            val lineCorrectlyDefined = (expectedLine.toInt until toLine).exists{ line =>
```

Comment:
```suggestion
            val lineCorrectlyDefined = (expectedLine until toLine).exists{ line =>
```

in addition to the change suggested by @dwijnand